docs/DeveloperGuide.md

```diff
@@ -1,34 +1,670 @@
+
+![Figure 4.2-1](images/DeveloperGuide/Figure%204.2-1.png)    
+_Figure 4.2-1: Sequence diagram for editDeadlineCommand execution_
+
+##### Design Considerations
+
+Due to the difference in the types of tasks and their parameters, this functionality was designed to guide the User
+ explicitly to ensure accurate inputs and thus efficiency. This will have minimal impact on the efficiency of the program,
+ while improving the quality of use for the User.
+ 
+### 4.3 Deleting of Lessons, Events, Deadlines (Fairuz)
+
+The `deleteTaskCommand` allows the user to delete any `Task` (lesson, event or deadline) from the `TaskList`.
+
+##### Implementation
+The User first enters the `deleteTask` command and appends the type of task to be deleted. **LifEasier** will then print
+ out the list of tasks of the intended type. For example, when User inputs `deleteTask /type event`, **LifEasier** will 
+ print out all `Events` in the `TaskList`.
+
+**LifEasier** will then prompt the user to select the `Task` to be deleted by inputting the index of the `Task` as seen from
+ the terminal. Before the `Task` is deleted, a temporary copy of the current state of the `Task` will be saved by 
+ `TaskHistory`. The `Task` is then deleted and **LifEasier** will print the confirmation of the deletion. The old copy of the 
+ `Task` will then be pushed to `TaskHistory`, and the updated `TaskList` will be saved by `Storage`. Figure 4.3 will illustrate 
+ the flow of the `deleteTaskCommand` through a sequence diagram.
+
+![Figure 4.3-1](images/DeveloperGuide/Figure%204.3-1.png)    
+_Figure 4.3-1: Sequence diagram for deleteTaskCommand execution_
+
+##### Design Considerations
+
+The command has to handle separate types of tasks as printing all tasks and forcing the User to look up the whole table
+ would be impractical in the long run. The enhanced capability with `TaskHistory` to allow the User to undo any action is 
+ crucial as its initial functionality did not have the ability to restore any accidentally deleted `Tasks`.
+
+### 4.4 Adding Notes (Edmund)
+
+The `addNotes` command adds user’s notes to the NoteList with a specified title and description. 
+
+##### Implementation
+
+The `addNotes` command first starts with 2 paths: with or without title. If a title is added alongside the 
+“addNotes” (ie. the user inputs “addNotes cats”) then the title is passed on to a method `isValidTitle` that would 
+check for a blank input. Conversely, if no title is appended, then the system would prompt the user for the title. 
+Either way, the system would still check for if the user inputs an empty title. 
+Next, the system would prompt the user for a description. Similar to title, the same method `isValidTitle`, to check 
+if the user has inputted an empty String.
+Finally, if both title and description are of valid input, then the 2 parameters would be added to the `NoteList` 
+and passed on to `NoteStorage`. 
+
```

Comment:
![image](https://user-images.githubusercontent.com/60388723/97529392-c4d0ec00-19ea-11eb-83c4-73babd363f34.png)
Would the prompt for the description be a call for a method instead of a return and the description the return value that has been prompted for?

Reply:
![image](https://user-images.githubusercontent.com/60292763/97656960-ab3fab00-1aa3-11eb-81c2-725f6dc167c9.png)

Should there be a main timeline? Perhaps the check valid methods can return a value to `:AddNotesCommand`. And is there a missing dashed line from `:Note` back to `:AddNotesCommand` to give back the control?